### PR TITLE
Prevent registration of usernames that conflict with URLs

### LINKED ISSUE #1171: Prevent using usernames that conflict with URLs, or make sure there are no conflicting URLs
It is currently possible to create a user with a username of `profile`.

If you login as that user and try to go to your user profile, you'll go into a redirect loop.

The culprit is here: https://github.com/azavea/nyc-trees/blob/a3f503f817c43df6d4fc2c4729942de7da48b275/src/nyc_trees/apps/users/urls/user.py#L12-L16

We should either 1. prevent usernames of `profile`, `achievements`, and `settings`, or 2. Change our URL patterns so using a username as a slug is unambiguous

### PR DESCRIPTION
**Pull request by kdeloach**
Add form validation error to prevent registration of usernames that
would conflict with any of our special `/user/*` URLs.

Fixes #1171

### CODE REVIEW

File: src/nyc_trees/apps/login/forms.py [hunk 1]
```diff
@@ -39,6 +40,12 @@ class NycRegistrationForm(RegistrationFormUniqueEmail):
         required=False
     )
 
+    def clean_username(self):
+        if self.cleaned_data['username'].lower() in User.reserved_usernames:
+            raise ValidationError('This username is invalid because it '
```
Comment (jwalgran) [inline src/nyc_trees/apps/login/forms.py]: This is a little specific and jargony for for a user facing error. "This username is already in use" is a white lie, but, perhaps, a bit friendlier.

Comment (kdeloach) [inline src/nyc_trees/apps/login/forms.py]: That thought crossed my mind too, will update.

Comment (kdeloach) [inline src/nyc_trees/apps/login/forms.py]: Settled on "This username is not available."

File: src/nyc_trees/apps/core/models.py [hunk 2]
```diff
@@ -100,6 +100,12 @@ class User(NycModel, AbstractUser):
                       'contributions_are_public': 'contributions',
                       'achievements_are_public': 'achievements'}
 
+    # Please keep in sync with special `/user/*` endpoints.
+    # Ref: src/nyc_trees/apps/users/urls/user.py
+    reserved_usernames = ('profile',
+                          'settings',
+                          'achievements')
```
Comment (kdeloach) [inline src/nyc_trees/apps/core/models.py]: Should this be a module-level constant?

**Comment by maurizi** [comment]
+1

**Pull request merged**